packages/ui5-task-pwa-enabler/lib/pwaEnabler.js

```diff
@@ -0,0 +1,286 @@
+        throw "Configuration is missing";
+    }
+    switch (configuration.serviceWorker.strategy) {
+        case swStrategies.offlinePage:
+            await addOfflinePageSw(configuration.serviceWorker);
+            break;
+        case swStrategies.offlineCopy:
+            await addOfflineCopySw();
+            break;
+        case swStrategies.offlineCopyWithBackup:
+            await addOfflineCopyBackupSw(configuration.serviceWorker);
+            break;
+        case swStrategies.preCache:
+            await addCacheFirstSw(configuration.serviceWorker);
+            break;
+        case swStrategies.advancedCache:
+            await addAdvancedCachingSw(configuration.serviceWorker);
+            break;
+        default:
+            throw "Unknown service worker strategy";
+    }
+    await addServiceworkerRegistration();
+    await addManifest(configuration.manifest);
+    return undefined;
+};
+
+/**
+ * Adds the script tag required to register a service worker to the index.html, the added code will register
+ * service-worker.js as the service worker for this website.
+ *
+ * @returns {Promise<undefined>}
+ */
+async function addServiceworkerRegistration() {
+    let fileContent = fs.readFileSync(path.join(__dirname, "../templates/register-sw.js"));
+    await appendToIndexHtmlHead("<script>" + fileContent.toString() + "</script>");
+    return undefined;
+}
+
+/**
+ * Writes a new file using the DuplexCollection provided by the caller of this module
+ *
+ * @param {object} parameters
+ * @param {string} parameters.path The path to the resource that should be written
+ * @param {string} parameters.content The content to write to the file
+ * @return {Promise} which resolves once the data has been written
+ */
+function writeFile({path, content}) {
+    return ws.write(new ui5Fs.Resource({path: path, string: content}));
+}
+
+/**
+ * Takes the template for the offlinePage service worker and fills it with the values provided in parameters
+ *
+ * @param {object} parameters
+ * @param {string} parameters.offlinePage Path to the file that will be used as offline page
+ * @returns {Promise<undefined>}
+ */
+async function addOfflinePageSw(parameters) {
+    await renderServiceWorker(path.join(__dirname, "../templates/offlineCopy.js"), parameters);
+    return undefined;
+}
+
+/**
+ * Copies the service worker template for the OfflineCopy service worker, no templating needs to be done
+ *
+ * @returns {Promise<undefined>}
+ */
+async function addOfflineCopySw() {
+    await renderServiceWorker(path.join(__dirname, "../templates/offlineCopy.js"), {});
+    return undefined;
+}
+
+/**
+ * Takes the template for the offlineCopyWithBackup service worker and fills it with the values provided in parameters
+ *
+ * @param {object} parameters
+ * @param {string} parameters.offlinePage Path to the file that will be used as offline page
+ * @returns {Promise<undefined>}
+ */
+async function addOfflineCopyBackupSw(parameters) {
```

Comment:
where are the parameters used - `offlinePage` is never used here, right?

Reply:
`offlinePage` is used here as a backup when a page hasn't been cached yet, the one without a backup page is called `offlineCopy`